### PR TITLE
Updating Fortran bindings with private helpers

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: src/umpire/interface/c_fortran/wrapResourceManager.cpp [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
 // project contributors. See the COPYRIGHT file for details.
 //
 // SPDX-License-Identifier: (MIT)
 #include "wrapResourceManager.h"
+#include <stdlib.h>
 #include <string>
 #include "umpire/Allocator.hpp"
 #include "umpire/ResourceManager.hpp"
 
-#include "umpire/strategy/AllocationAdvisor.hpp"
```
Comment (davidbeckingsale) [inline src/umpire/interface/c_fortran/wrapResourceManager.cpp:12]: @ltaylor16 - I'm not sure why these lines get removed, but now this fails to compile...

**Comment by ltaylor16** [comment]
It looks like an older version of Shroud was used.  The old files say Shroud 0.11.0 and the new files do not have a version but should say 0.12.1

**Comment by davidbeckingsale** [comment]
> It looks like an older version of Shroud was used. The old files say Shroud 0.11.0 and the new files do not have a version but should say 0.12.1

Wow, I really need to learn how to install Python tools... thanks Lee, sorry for bothering you.

**Comment by ltaylor16** [comment]
Looking at the changes, there are some derived types which need to be private since they may be different depending on which version of Shroud generated the files.
Let me update Shroud before you merge.

**Pull request merged**